job_executor/worker/steps/dataset_pseudonymizer.py

```diff
@@ -20,192 +22,131 @@ def _get_unit_types(
+    """
+    return [
+        name for name in dataset.schema.names if name not in excluded_columns
+    ]
 
 
-def _pseudonymize_identifier_and_measure(
-    input_csv_path: Path,
-    identifier_unit_id_type: str,
-    measure_unit_id_type: str,
+def _pseudonymize(
+    input_parquet_path: Path,
+    identifier_unit_id_type: Optional[str],
+    measure_unit_id_type: Optional[str],
     job_id: str,
 ) -> Path:
-    unique_idents = set()
-    unique_measure_values = set()
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            row = line.strip().split(";")
-            unit_id = row[1]
-            value = row[2]
-            unique_idents.add(unit_id)
-            unique_measure_values.add(value)
-    identifier_to_pseudonym = pseudonym_service.pseudonymize(
-        list(unique_idents), identifier_unit_id_type, job_id
+    input_dataset = dataset.dataset(input_parquet_path)
+
+    unit_id_pseudonyms = _pseudonymize_if_needed(
+        input_dataset, "unit_id", identifier_unit_id_type, job_id
     )
-    value_to_pseudonym = pseudonym_service.pseudonymize(
-        list(unique_measure_values), measure_unit_id_type, job_id
+    value_pseudonyms = _pseudonymize_if_needed(
+        input_dataset, "value", measure_unit_id_type, job_id
     )
-    output_csv_path = (
-        input_csv_path.parent / f"{input_csv_path.stem}_pseudonymized.csv"
+
+    excluded_columns = []
+    if unit_id_pseudonyms:
+        excluded_columns.append("unit_id")
+    if value_pseudonyms:
+        excluded_columns.append("value")
+    column_names = _get_columns_excluding(input_dataset, excluded_columns)
+
+    unprocessed_columns = input_dataset.to_table(columns=column_names)
+
+    arrays_to_include = [
+        col
+        for col in [unit_id_pseudonyms, value_pseudonyms]
+        if col is not None
+    ]
+    arrays_to_include.extend(
+        [unprocessed_columns[name] for name in column_names]
     )
-    target_file = open(output_csv_path, "w", newline="", encoding="utf-8")
-    with open(input_csv_path, newline="", encoding="utf-8") as csv_file:
-        for line in csv_file:
-            row = line.strip().split(";")
-            line_number: int = row[0]
-            unit_id: str = row[1]
-            value: str = row[2]
-            start_date: str = row[3]
-            stop_date: str = row[4]
-            target_file.write(
-                ";".join(
-                    [
-                        str(line_number),
-                        str(identifier_to_pseudonym[unit_id]),
-                        str(value_to_pseudonym[value]),
-                        start_date,
-                        stop_date,
-                    ]
-                )
-                + "\n"
-            )
-    target_file.close()
-    return output_csv_path
 
+    # Construct the pseudonymized table
+    pseudonymized_table = pyarrow.Table.from_arrays(
+        arrays_to_include, input_dataset.schema.names
```

Comment:
I have not considered this before, but I think this will actually break and mess up the column order if the only column being pseudonymized is "value" 🤔 Can we make a unit test for this?